src/Ship.cpp

```diff
@@ -361,6 +364,87 @@ void Ship::UpdateMass()
 	SetMass((m_stats.static_mass + GetPropulsion()->FuelTankMassLeft()) * 1000);
 }
 
+template <typename T>
+inline int sign(T num)
+{
+	return (num >= 0) - (num < 0);
+}
+
+vector3d Ship::CalcAtmosphericForce() const
+{
+	// Data from ship.
+	double m_topCrossSec = GetShipType()->topCrossSection;
+	double m_sideCrossSec = GetShipType()->sideCrossSection;
+	double m_frontCrossSec = GetShipType()->frontCrossSection;
+
+	// TODO: vary drag coefficient based on Reynolds number, specifically by
+	// atmospheric composition (viscosity) and airspeed (mach number).
+	double m_topDragCoeff = GetShipType()->topDragCoeff;
+	double m_sideDragCoeff = GetShipType()->sideDragCoeff;
+	double m_frontDragCoeff = GetShipType()->frontDragCoeff;
+
+	double m_shipLiftCoeff = GetShipType()->shipLiftCoefficient;
+
+	// By converting the velocity into local space, we can apply the drag individually to each component.
+	auto m_localVel = GetOrient().Inverse() * GetVelocity();
+	auto m_lVSqr = m_localVel.LengthSqr();
+
+	// The drag forces applied to the craft, in local space.
+	// TODO: verify dimensional accuracy and that we're not generating more drag than physically possible.
+	// TODO: use a different drag constant for each side (front, back, etc).
+	// This also handles (most of) the lift due to wing deflection.
+	vector3d fAtmosDrag = vector3d(
+		CalcAtmosphericDrag(m_lVSqr, m_sideCrossSec, m_sideDragCoeff),
```

Comment:
I think that if `CalcAtmosphericDrag` will return a `vector3d` and accept as second and third elements a `vector3d`
then you could avoid to treble calls and code become more readable.

Reply:
I think overloading the function to accept `vector3d` would be best suited in this case, and it'd make things a little easier to read.